vtysh/vtysh_config.c

```diff
@@ -79,29 +81,50 @@ static void config_del(struct config *config)
 	XFREE(MTYPE_VTYSH_CONFIG, config);
 }
 
+static int config_cmp(const struct config *c1, const struct config *c2)
+{
+	return strcmp(c1->name, c2->name);
+}
+
+static uint32_t config_hash(const struct config *c)
+{
+	return string_hash_make(c->name);
+}
+
 DECLARE_LIST(config_master, struct config, rbt_item)
+DECLARE_HASH(config_master_hash, struct config, hash_item, config_cmp,
+	     config_hash)
+
+/*
+ * The config_master_head is a list for order of receipt
+ * The hash is for quick lookup under this NODE
+ */
+struct configuration {
+	struct config_master_head master;
+	struct config_master_hash_head hash_master;
+};
 
 static struct config *config_get(int index, const char *line)
 {
 	struct config *config, *config_loop;
-	struct config_master_head *master;
+	struct configuration *configuration;
+	struct config lookup;
 
 	config = config_loop = NULL;
 
-	master = vector_lookup_ensure(configvec, index);
+	configuration = vector_lookup_ensure(configvec, index);
 
-	if (!master) {
-		master = XMALLOC(MTYPE_VTYSH_CONFIG, sizeof(struct config_master_head));
-		config_master_init(master);
-		vector_set_index(configvec, index, master);
+	if (!configuration) {
+		configuration = XMALLOC(MTYPE_VTYSH_CONFIG,
+					sizeof(struct configuration));
+		config_master_init(&configuration->master);
+		config_master_hash_init(&configuration->hash_master);
+		vector_set_index(configvec, index, configuration);
 	}
 
-	frr_each (config_master, master, config_loop) {
-		if (strcmp(config_loop->name, line) == 0) {
-			config = config_loop;
-			break;
-		}
-	}
+	lookup.name = XSTRDUP(MTYPE_VTYSH_CONFIG_LINE, line);
+	config = config_master_hash_find(&configuration->hash_master, &lookup);
```

Comment:
PR looks good to me, but the `XSTRDUP` above is probably unnecessary since the hash lookup won't modify the provided string.

Reply:
I cannot assign lookup.name = line because line is const.  I cannot make lookup.name const either

Reply:
In that case you can just cast away the constness to make the compiler happy (which is safe in this case). Example:
```
lookup.name = (char *)line;
```

Reply:
fixed